src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -167,25 +184,32 @@ private bool TrimMethods(ITaskItem assemblyItem)
         if (isTrimmed)
         {
             AddItemToTrimmedList(assemblyFilePathArg, trimmedAssemblyFilePath);
+            newAssmeblyItem.ItemSpec = trimmedAssemblyFilePath;
         }
 
+        _updatedAssemblies.Add(newAssmeblyItem);
         return true;
     }
 
-    private static string ComputeTrimmedAssemblyPath(string assemblyFilePath)
+    private static string ComputeTrimmedAssemblyFolderName(string assemblyFilePath)
     {
         string? assemblyPath = Path.GetDirectoryName(assemblyFilePath);
-        string? assemblyName = Path.GetFileNameWithoutExtension(assemblyFilePath);
         if (string.IsNullOrEmpty(assemblyPath))
         {
-            return (assemblyName + "_trimmed.dll");
+            return "trimmedAssemblies";
         }
         else
         {
-            return Path.Combine(assemblyPath, (assemblyName + "_trimmed.dll"));
+            return Path.Combine(assemblyPath,  "trimmedAssemblies");
```

Comment:
All of the trimmed assemblies should go into one directory, which is in our intermediate output directory.

The above would fail or be bad if the assembly path happened to be from a nuget package, or the runtime pack. Similar to the no-trimming test that was failing.

Reply:
ILLinker uses a `linked/` for example.

Reply:
Are you suggesting `trimmed` instead of `trimmedAssemblies`?

Reply:
I wasn't suggesting what the folder name should be - `trimmed` would be preferable, but either is fine really.

I was saying that with the current code the `trimmedAssemblies` directory is created next to the original assembly itself, which be a nuget cache path, or a runtime pack path etc. Instead of that, create one `trimmed` or `trimmedAssemblies` path in `obj` (WasmIntermediateOutputPath), and emit all the trimmed assemblies there. Easy to clean up too.

Reply:
Fixed.